cit-api/pipeline/views/opportunity.py

```diff
@@ -42,14 +57,98 @@ def get_queryset(self):
             if(regional_district_models is not None):
                 queryset = queryset.filter(geo_position__intersects=regional_district_models.geom)
 
+        exclude_unknowns = self.request.query_params.get('exclude_unknowns', None)
+        queryset = self.service_queryset(queryset, exclude_unknowns, 'opportunity_road_connected')
+        queryset = self.service_queryset(queryset, exclude_unknowns, 'opportunity_water_connected')
+        queryset = self.service_queryset(queryset, exclude_unknowns, 'opportunity_sewer_connected')
+        queryset = self.service_queryset(queryset, exclude_unknowns, 'opportunity_electrical_connected')
+        queryset = self.service_queryset(queryset, exclude_unknowns, 'opportunity_natural_gas_connected')
+
+        # TODO Figure out distance calulation issues
+        community_id = int(self.request.query_params.get('community_id', INVALID_INT))
+        community_distance = float(self.request.query_params.get('community_distance', INVALID_INT))
+        if(community_distance >= MIN_DISTANCE and community_id >= MIN_TABLE_ID):
+            community_model = Community.objects.get(id=community_id)
+            if(community_model is not None):
+                queryset = queryset.filter(geo_position__distance_lte=(community_model.point,
+                                                                       D(km=community_distance)))
+        
+        parcel_size_min = float(self.request.query_params.get('parcel_size_min', INVALID_INT))
+        parcel_size_max = float(self.request.query_params.get('parcel_size_max', INVALID_INT))
+        if(parcel_size_min >= MIN_SIZE):
+            queryset = queryset.filter(parcel_size__gte=parcel_size_min)
+        if(parcel_size_max >= MIN_SIZE):
+            queryset = queryset.filter(parcel_size__lte=parcel_size_max)
+
+        zoning = self.request.query_params.get('zoning', None)
+        if(zoning is not None):
+            zonings = zoning.split(',')
+            queryset = queryset.filter(Q(land_use_zoning__in=zonings) | Q(ocp_zoning_code__in=zonings))
+
+        return queryset
+
+    def service_queryset(self, queryset, exclude_unknowns, service_name):
+        service_connected = self.request.query_params.get(service_name, None)
+        print("----------")
```

Comment:
Do we want to get rid of these print statements before merging?

Reply:
I agree that we should remove these, but I'll be taking them out in an upcoming PR and I think they're fine to be in develop until then.

Reply:
Okay sounds good!